packages/plugin-betswirl/src/actions/coinToss.ts

```diff
@@ -0,0 +1,233 @@
+import { z } from "zod";
+import { type Hex } from "viem";
+import {
+    type Action,
+    type IAgentRuntime,
+    type Memory,
+    type HandlerCallback,
+    type State,
+    elizaLogger,
+    type ActionExample,
+    composeContext,
+    generateObject,
+    ModelClass,
+} from "@moxie-protocol/core";
+import { MoxieWalletClient } from "@moxie-protocol/moxie-agent-lib/src/wallet";
+import {
+    CASINO_GAME_TYPE,
+    CoinToss,
+    COINTOSS_FACE,
+    slugById,
+    formatTxnUrl,
+} from "@betswirl/sdk-core";
+import { casinoBetParams, getMaxBetCountParam } from "../types";
+import {
+    getChainIdFromWallet,
+    getBetToken,
+    placeBet,
+    getBet,
+    getBetAmountInWei,
+} from "../utils/betswirl";
+import { formatTokenForMoxieTerminal } from "../utils/moxie";
+
+export const CoinTossBetParameters = z.object({
+    face: z.nativeEnum(COINTOSS_FACE).describe("The face of the coin"),
+    ...casinoBetParams,
+    ...getMaxBetCountParam(CASINO_GAME_TYPE.COINTOSS),
+});
+export const coinTossTemplate = `
+Extract the following details to flip a coin:
+- **betAmount** (String): The amount to wager.
+- **face** (String): The side of the coin to bet on. Can be either:
+  - HEADS
+  - TAILS
+- **token** (String): The optional token symbol.
+
+Provide the values in the following JSON format:
+
+\`\`\`json
+{
+    "betAmount": string,
+    "face": string,
+    "token": string
+}
+\`\`\`
+
+Here are example messages and their corresponding responses:
+
+**Message 1**
+
+\`\`\`
+Bet 0.01 ETH on heads
+\`\`\`
+
+**Response 1**
+
+\`\`\`json
+{
+    "betAmount": "0.01",
+    "face": "heads",
+    "token" "ETH"
+}
+\`\`\`
+
+**Message 2**
+
+\`\`\`
+Double or nothing 0.5 on heads
+\`\`\`
+
+**Response 2**
+
+\`\`\`json
+{
+    "betAmount": "0.5",
+    "face": "HEADS",
+    "token": "",
+}
+\`\`\`
+
+Here are the recent user messages for context:
+{{recentMessages}}
+`;
+export const coinTossAction: Action = {
+    name: "COIN_TOSS",
+    similes: [
+        "COIN_FLIP",
+        "DOUBLE_OR_NOTHING",
+        "TOSS_A_COIN",
+        "BETSWIRL_COIN_TOSS",
+    ],
+    description:
+        "Flip a coin on BetSwirl. The player is betting that the rolled face will be the one chosen.",
+    suppressInitialMessage: true,
+    validate: async (
+        _runtime: IAgentRuntime,
+        _message: Memory,
+        _state: State
+    ) => true,
+    handler: async (
+        runtime: IAgentRuntime,
+        message: Memory,
+        state: State,
+        _options: { [key: string]: unknown },
+        callback: HandlerCallback
+    ) => {
+        try {
+            elizaLogger.log("Starting COIN_TOSS handler...");
+
+            // Validate the chain
+            const wallet = state.agentWallet as MoxieWalletClient;
```

Comment:
Minor change from Moxie client, just need to commit this change

```suggestion
            const wallet = state.moxieWalletClient as MoxieWalletClient;
```